File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    a = []
    b = []
    # 큰 숫자가 앞에 오도록 순서 변경
    for x, y in sizes:
        if x < y:
            x, y = y, x
        a.append(x)
        b.append(y)
        
    return max(a) * max(b)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int[] w = new int[sizes.length];
        int[] h = new int[sizes.length];
        
        int idx = 0;
        for (int[] pair : sizes) {
            w[idx] = Math.max(pair[0], pair[1]);
            h[idx++] = Math.min(pair[0], pair[1]);
        }
        
        Arrays.sort(w);
        Arrays.sort(h);
        
        return w[idx-1] * h[idx-1];
    }
}